manager.py

```diff
@@ -79,28 +80,103 @@ def stop(self):
         self._rest_manager.remove_web_handler(self._handler)
         super().stop()
 
-    def get_blocks_structure(self):
-        """ Return file structure under 'blocks' resource """
-        blocks = []
+    def get_blocks_structure(self, branches = 0):
+        """ Get block structure from disk
+
+        Args:
+            branches: Show git branches for each block
+
+        Returns:
+           Structure with an Array of blocks
+        """
+
+        # Return
+        ret = []
+
+        # Blocks from path
         blocks_paths = self._get_abs_blocks_paths()
+
+        # Each block
         for blocks_path in blocks_paths:
             for item in listdir(blocks_path):
-                # discard known exclusions
+
+                # -- Discard known exclusions
                 if item in ProjectManager.exclusions:
                     continue
 
-                item_full_path = path.join(blocks_path, item)
+                # -- Make block full path
+                block_full_path = path.join(blocks_path, item)
+
+                # -- Folder or file
+                block = None
+                if path.isdir(block_full_path):
+                    block = {"name": item, "type": "directory"}
+                elif path.isfile(block_full_path):
+                    block = {"name": path.splitext(item)[0], "type": "file"}
+
+                # -- Get branches
+                if branches:
+                    block["branch"] = self._get_git_branch(block_full_path)
+
+                # -- Add block
+                if block is not None:
+                    ret.append(block)
+
+
+        return {"blocks": ret}
+
+    def _get_git_branch(self, directory):
+        """Get Git branch
+
+        Args:
+            path_to_block: path to block to get Git branch for
+
+
+        Returns:
+           String of branch name or None
+
+        """
+
+        # Command
+        cmd = "cd %s && git rev-parse --abbrev-ref HEAD" % directory
+
+        # Get git branch name
+        result = os.popen(cmd).read().strip()
 
-                if path.isdir(item_full_path):
-                    blocks.append({"name": item,
-                                   "type": "directory"})
-                elif path.isfile(item_full_path):
-                    blocks.append({"name": path.splitext(item)[0],
-                                   "type": "file"})
+        # Result
+        if len(result):
+            return result
+        return None
```

Comment:
I wonder if this can be improved somehow, would there be a case when no branch is retrieved?, if there is, can we provide a description instead of setting it to None, should something be logged?